src/iocore/net/BIO_fastopen.cc

```diff
@@ -21,12 +21,117 @@
  *  limitations under the License.
  */
 
+#include <openssl/opensslv.h>
+
 #include "P_Net.h"
 #include "iocore/eventsystem/SocketManager.h"
 #include "tscore/ink_assert.h"
+#include "tscore/ink_config.h"
 
 #include "BIO_fastopen.h"
 
+#if defined(BORINGLIKE)
+#error
+#elif defined(OPENSSL_IS_BORINGSSL) || defined(OPENSSL_IS_AWSLC)
+#define BORINGLIKE 1
+#else
+#define BORINGLIKE 0
+#endif
+
+namespace
+{
+
+#if defined(HAVE_BIO_GET_EX_NEW_INDEX) && defined(HAVE_BIO_GET_EX_DATA) && defined(HAVE_BIO_SET_EX_DATA)
+
+class ExData
+{
+public:
+  // Pseudo-namespace
+  ExData() = delete;
+
+  static int
+  idx()
+  {
+    static int idx_ = []() -> int {
+      int i = BIO_get_ex_new_index(0, nullptr, _new, _dup, _free);
+      ink_release_assert(i >= 0);
+      return i;
+    }();
+    return idx_;
+  }
+
+private:
+#if BORINGLIKE
+  static const CRYPTO_EX_unused _new{};
+  // Hopefully Boring intializes ex data pointers to nullptr by default.
```

Comment:
How about setting `nullptr` in `fastopen_create` by `BIO_set_ex_data`? I'm not sure if this is really possible. It could be too early to call `BIO_set_ex_data`.

Reply:
Another option is setting nullptr after making the BIO on SSLNetVC side. It'd be a silly requirement if this BIO is publicly available for others though.

Reply:
I made a change in SSLNetVConnection.cc to initialize the ex data to either the destination address or nullptr for all BIOs using the fastopen methods.